### ISSUE TITLE
<video> element is not appearing in Safari

### ISSUE THREAD

**Issue by jonyheim** [wottpal/kirby-video#1]
I've tried to use the plugin on multiple pages, but for some reason the video does not get embedded in Safari, but it is working fine in Chrome. Here are two links where I'm using the plugin:

- http://jony.io/portfolio/1rm
- http://jony.io/blog/1rm-the-retrospective

Please note, these pages are still "invisible," so some of the links haven't been mapped yet (i.e. download on the App Store).

**What I've Done**
- I added the `kirby-video-master` folder to my site under `/plugins`, and renamed the folder to `video`.
- Added `<?php echo css('assets/plugins/video/video.min.css') ?>` (I also used the snippet you provided in the Readme, which does the same thing).
- Added `<?= js('assets/plugins/video/video.min.js') ?>` right before the closing `</body>` tag.
- Exported the video as a .mp4 @ 720p and created an image with the same dimensions (I'll add these at the end of this under a section titled Assets).
- In the post, I'm using `(video: 1rm-loop preset: gif)` (which is the functionality I'm looking for), but I've also tested removing the preset, and using the `options` tag instead. 

**Tech**
My site is hosted on MediaTemple, and is running PHP version 7.0.6 CGI. Safari is running v10.1.2, and my Mac is on Sierra 10.12.6. 

**Assets**
- [1rm-loop.png](https://user-images.githubusercontent.com/15092804/29243854-89853130-7f5e-11e7-8207-a95ead8bf1d2.png)
- [1rm-loop.mp4.zip](https://github.com/wottpal/kirby-video/files/1220032/1rm-loop.mp4.zip)

Here's a screenshot of the Kirby panel for the first link as well. Thanks in advance for all your help! Hopefully this is enough information to figure this out. 🍻 

![screen shot 2017-08-12 at 1 04 14 pm](https://user-images.githubusercontent.com/15092804/29243876-0d02f9b6-7f5f-11e7-978c-c4a1e3086d04.png)

**Comment by wottpal** [comment]
Hi Jony,
first I can confirm that your site is working for me in Chrome but not in Safari, too. But I've tested the plugin with your files on my site and everything works as expected, that's odd. 

I've listed some things I would like you to try and then tell me about the outcomes:

0. Clear Cache on Safari (CMD-ALT-E when Dev-Mode is enabled)
1. Embed the video without `preset: gif` and lazyload enabled, like `(video: 1rm-loop options: controls)` and look which children (`<source>`'s) the `<video>` element in the DOM has.
2. Try to embed the video on a completely empty page (no other content)
3. Try to use the [attached mp4 file](https://www.dropbox.com/s/ylxn0dgvkb38wan/1rm-loop.mp4?dl=0) instead (I've just re-saved it with [Handbrake](https://handbrake.fr))
4. Try to delete `1rm-loop.png` to let the plugin just use the mp4 file
5. Try to rename both `1rm-loop.mp4` to something else

Sorry for all these (maybe silly) ideas, hopefully we can somehow drill down on your problem!

Cheers 🍻
Dennis

**Comment by jonyheim** [comment]
Definitely not silly; I was hoping it might be one of those as well. I've tried what you suggested, but so far no dice. For #1, here's what the DOM was returning:

``` html
<!-- This is what Chrome is showing when I use "options: controls" -->

<figure class="html-video " style="color:#fff;height: 720pxwidth:960 px" loaded="true">  
  <!-- Video -->
  <video controls="" width="100%">

          <source src="http://jony.io/content/video-test/videotest.mp4" type="video/mp4">

          <img src="http://jony.io/content/video-test/videotest.png">

  </video>

</figure>

<!-- and here is what Safari is showing -->

<figure class="html-video " style="color:#fff;height: 720pxwidth:960 px" loaded="true">  
  <!-- Video -->
  <img src="http://jony.io/content/video-test/videotest.png">

</figure>
```

This time around, I noticed something different. When I load the page for the first time in Safari, the image isn't loading either. Only the white background is showing. When you refresh the page, the image gets displayed.

``` html
<!-- Safari is only showing the white background until refresh. Looks like "loaded=true" isn't working on first load. -->

<figure class="html-video " style="color:#fff;height: 720pxwidth:960 px">  
  <!-- Video -->
  <img src="http://jony.io/content/video-test/videotest.png">

</figure>
```

I've gone ahead and tried a few more things that I could think of, but nothing seems to be working so far. Here's what I tried:

- Removing all CSS but the link to the plugin's css file.
- Removing all javascript, but the link to the plugin's js file.
- Downloading the site to a localhost to make sure it wasn't an issue with the server.
- Updating the site to Kirby 2.5.4

Here's a link to the page with no content, but the `.mp4` and `.png` files: http://jony.io/video-test

Also, (to cover all bases) these are the files as they appear on the server:

<img width="222" src="https://user-images.githubusercontent.com/15092804/29253347-c3274d26-802e-11e7-8524-de4ae4353d5d.png">

Knowing my luck, there's a typo somewhere, but I don't see any errors in the web inspector. Very strange considering the files worked fine on your end. 😞

**Comment by jonyheim** [comment]
I did some more snooping – Happy Sunday! 

I don't think this is an error with your plugin, since I can't even seem to load the video in Safari when I go directly to http://jony.io/content/video-test/videotest.mp4 – would you mind trying to load this in your browser to see if it loads? 

Strangely enough, I also tried using your `demo.mp4` file that was included in the plugin folder, but I'm getting the same error. http://jony.io/content/video-test/demo.mp4

<img width="1440" alt="screen shot 2017-08-13 at 2 49 03 pm" src="https://user-images.githubusercontent.com/15092804/29253618-a197c264-8036-11e7-9dac-bd9eed261945.png">


<img width="1119" alt="screen shot 2017-08-13 at 2 45 44 pm" src="https://user-images.githubusercontent.com/15092804/29253590-1e1e6014-8036-11e7-93d6-29e14adf289f.png">

In the Web Inspector Under Network > All Resources, Safari is returning "An error occurred trying to load the resource." 

![screen shot 2017-08-13 at 2 47 42 pm](https://user-images.githubusercontent.com/15092804/29253606-6352f032-8036-11e7-8646-d4427c343bca.png)


So strange. Not sure how to go about this one…

**Comment by wottpal** [comment]
Hey @jonyheim,
sorry for getting back so late. Had a busy week :)

I've uploaded your file to my server and it loads fine in Safari: http://wottpal.com/files/videotest.mp4. Yours doesn't load for me neither. So it definitely seems like a server error on your side. 

Do you run your site on your own custom server? Apache or nginx? Unfortunately, I only have some experience with nginx because I use it by my own. I've attached my `/etc/nginx/mime.types` file just for reference because thats where I could imagine the error in your configuration. And don't forget to restart nginx after you made any changes with something like `sudo service nginx restart`. If that doesn't help maybe have a look into your `/etc/nginx/sites-enabled/` or re-install nginx completely.

[mime.types.zip](https://github.com/wottpal/kirby-video/files/1237034/mime.types.zip)

If you use Apache make sure to use exactly the `.htaccess` file provided by Kirby and maybe it needs special permissions.

**Comment by jonyheim** [comment]
@wottpal Thanks again for your help. I downloaded a fresh install of Kirby onto my server, and it works. I think there might be an issue with my "asset firewall" and plugins being able to access files. I've created a thread on the Kirby forum, so hopefully I'll be able to get to the bottom of this 😅

https://forum.getkirby.com/t/would-the-asset-firewall-block-plugins-from-accessing-videos/

Your plugin is perfect btw! Just need to figure out how to get it to work why my installation 😄
